docs/source/command_line.rst

```diff
@@ -482,6 +482,32 @@ of the above sections.
            # 'items' now has type List[List[str]]
            ...
 
+.. option:: --local-partial-types
+
+    By default, mypy won't check partial types spanning module top level or class top level.
```

Comment:
This sentence is pretty vague. It would be better to also introduce the concept "partial types" before discussing it.

This is an important case where this option makes a difference:

```py
class A:
    x = None  # Need type annotation here if using --local-partial-types

    def __init__(self) -> None:
        self.x = 1

reveal_type(A().x)  # Union[int, None] without --local-partial-types
```

This example also makes it easier to see why this is called "local partial types" -- it disallows inferring variable type from two assignments in different scopes (for example, in class body and inside a method).